aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -264,31 +265,50 @@ static ScalarType get_dtype(Tensor& result, const Tensor& self, optional<ScalarT
   return src_type;
 }
 
-Tensor& sum_out(Tensor& result, const Tensor& self, IntArrayRef dim,
+Tensor& sum_out(Tensor& result, const Tensor& self, c10::optional<IntArrayRef> opt_dim,
```

Comment:
How BC-breaking is this? 

Will it break all C++ API users?
Would it be a silent behavior change for people who were passing `dims=()`?

If the latter, should we throw a runtime error for a release when people pass `dims=()`?